pkg/server/configuration.go

```diff
@@ -48,9 +58,119 @@ func (s *server) parseExtVars(unparsed interface{}) (map[string]string, error) {
 	return extVars, nil
 }
 
+func (s *server) parseFormattingOpts(unparsed interface{}) (formatter.Options, error) {
+	newOpts, ok := unparsed.(map[string]interface{})
+	if !ok {
+		return formatter.Options{}, fmt.Errorf("unsupported settings value for formatting. expected json object. got: %T", unparsed)
+	}
+
+	opts := formatter.DefaultOptions()
+	var (
+		valOpts = reflect.ValueOf(&opts).Elem()
+		typOpts = valOpts.Type()
+
+		typBool         = reflect.TypeOf(false)
+		typInt          = reflect.TypeOf(int(0))
+		typStringStyle  = reflect.TypeOf(formatter.StringStyleDouble)
+		typCommentStyle = reflect.TypeOf(formatter.CommentStyleHash)
+	)
```

Comment:
Instead of all that reflection, you should be able to use mapstructure in a pretty straightforward way: https://github.com/mitchellh/mapstructure

The comment style, string style etc can probably be handled by a decoder hook (a bit in the same way you're currently doing)

Reply:
yeah, I wasn't sure whether adding a new dependency was worth it, but the reflection approach didn't feel completely right either. I'll update the PR with mapstructure and see how it looks.